src/data/index.ts

```diff
@@ -0,0 +1,25 @@
+import { getCurrentSemester } from './api';
+
+export interface Fraction {
+	numerator: number;
+	denominator: number;
+}
+
+export interface CourseItem {
+	name: string;
+	weight: number;
+	grade?: number | Fraction;
+}
+
+export interface Course {
+	name: string;
+	crn?: number;
+	items: CourseItem[];
+}
+
+export interface Semester {
+	name: string;
+	courses: Course[];
+}
+
+export { getCurrentSemester };
```

Comment:
In the case of exporting all the members in this file it might be preferable to write:

```
export default {
  Fraction,
  CourseItem,
  Course,
  Semester
};
```

As it will make the import a bit nicer. Unless we would prefer to be importing these individually, in that case we might prefer to limit our number of interfaces to one per file.

Reply:
I did remove this line, and changed line 1 to `export { getCurrentSemester } from api.ts`. Also, the reason I don't export everything together like what you mentioned is so when people are modifying things, it is obvious that it is exported, and hopefully they understand that side effects can extend beyond the current file.